<a href="https://colab.research.google.com/github/nihdin/AppliedEconometrics/blob/master/AEA1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
install.packages("wooldridge")
library(wooldridge)

Installing package into ‘/usr/local/lib/R/site-library’
(as ‘lib’ is unspecified)



In [2]:
head(wage2)
attach(wage2)

wage hours IQ  KWW educ exper tenure age married black south urban sibs
1  769 40     93 35  12   11     2     31  1       0     0     1      1  
2  808 50    119 41  18   11    16     37  1       0     0     1      1  
3  825 40    108 46  14   11     9     33  1       0     0     1      1  
4  650 40     96 32  12   13     7     32  1       0     0     1      4  
5  562 40     74 27  11   14     5     34  1       0     0     1     10  
6 1400 40    116 43  16   14     2     35  1       1     0     1      1  
  brthord meduc feduc lwage   
1  2       8     8    6.645091
2 NA      14    14    6.694562
3  2      14    14    6.715384
4  3      12    12    6.476973
5  6       6    11    6.331502
6  2       8    NA    7.244227

In [3]:
colnames(wage2)

[1] "wage"    "hours"   "IQ"      "KWW"     "educ"    "exper"   "tenure" 
 [8] "age"     "married" "black"   "south"   "urban"   "sibs"    "brthord"
[15] "meduc"   "feduc"   "lwage"

In [4]:
mean_wage<-mean(wage)
cat("The mean wage of the sample is ",mean_wage, "\n")
mean_IQ<-mean(IQ)
cat("The mean IQ of the sample is ", mean_IQ, "\n")
cat("The standard deviation of IQ for the sample is ", sd(IQ), "\n")

The mean wage of the sample is  957.9455 
The mean IQ of the sample is  101.2824 
The standard deviation of IQ for the sample is  15.05264 


In [5]:
m1<-lm(wage~IQ)
m1
summary(m1)


Call:
lm(formula = wage ~ IQ)

Coefficients:
(Intercept)           IQ  
    116.992        8.303  



Call:
lm(formula = wage ~ IQ)

Residuals:
   Min     1Q Median     3Q    Max 
-898.7 -256.5  -47.3  201.1 2072.6 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)    
(Intercept) 116.9916    85.6415   1.366    0.172    
IQ            8.3031     0.8364   9.927   <2e-16 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 384.8 on 933 degrees of freedom
Multiple R-squared:  0.09554,	Adjusted R-squared:  0.09457 
F-statistic: 98.55 on 1 and 933 DF,  p-value: < 2.2e-16


In [6]:
cat("Sample size =",nrow(wage2),"\n")

Sample size = 935 


#The Linear regression model

##### **_$wage$_**= 116.992 + 8.303 **_$IQ$_**

#####$n$ =935
#####$R^{2}$ = 0.0955

The increase in wage for an increase in IQ of 15 points will increase the predicted salary by $124.545 (8.303*15). R-squared value is less than 10% from the model. Therefore we can mention that IQ score does not explain most of the variation in wage

In [7]:
m2<-lm(log(wage)~IQ)
m2
summary(m2)


Call:
lm(formula = log(wage) ~ IQ)

Coefficients:
(Intercept)           IQ  
   5.886994     0.008807  



Call:
lm(formula = log(wage) ~ IQ)

Residuals:
     Min       1Q   Median       3Q      Max 
-2.09324 -0.25547  0.02261  0.27544  1.21486 

Coefficients:
             Estimate Std. Error t value Pr(>|t|)    
(Intercept) 5.8869943  0.0890206   66.13   <2e-16 ***
IQ          0.0088072  0.0008694   10.13   <2e-16 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 0.3999 on 933 degrees of freedom
Multiple R-squared:  0.09909,	Adjusted R-squared:  0.09813 
F-statistic: 102.6 on 1 and 933 DF,  p-value: < 2.2e-16


#The Logarithmic model

##### **_$log(wage)$_**= 5.8869 + 0.0088 **_$IQ$_**

#####$n$ =935
#####$R^{2}$ = 0.0991

For change in IQ of 15 points, change in log(wage) = 0.132 (0.0088*15). Therefore there is an increase of 13.2 % increase in wage for an increase in IQ of 15 points 

#The Log model

##### **_$log(wage)$_**= 5.8869 + 0.0088 **_$IQ$_**

#####$n$ =935
#####$R^{2}$ = 0.0991

For change in IQ of 15 points, change in log(wage) = 0.132 (0.0088*15). Therefore there is an increase of 13.2 % increase in wage for an increase in IQ of 15 points 

In [0]:
attach(countymurders)

In [9]:
summary(countymurders)
cm<-countymurders[which(countymurders$year==1996),]
summary(cm)
cat("Current number of rows (Counties) = ",nrow(cm), "\n")

    arrests            countyid        density             popul        
 Min.   :   0.000   Min.   : 1001   Min.   :    0.05   Min.   :     85  
 1st Qu.:   0.000   1st Qu.:20105   1st Qu.:   17.68   1st Qu.:  13144  
 Median :   1.000   Median :36065   Median :   44.24   Median :  28792  
 Mean   :   6.782   Mean   :32922   Mean   :  252.24   Mean   :  89344  
 3rd Qu.:   3.000   3rd Qu.:48049   3rd Qu.:  106.60   3rd Qu.:  66480  
 Max.   :2391.000   Max.   :56045   Max.   :54058.77   Max.   :9127751  
 NA's   :504                                                            
    perc1019        perc2029        percblack         percmale    
 Min.   : 7.08   Min.   : 5.617   Min.   : 0.000   Min.   :35.15  
 1st Qu.:14.32   1st Qu.:12.302   1st Qu.: 0.200   1st Qu.:40.90  
 Median :15.42   Median :14.270   Median : 1.450   Median :41.81  
 Mean   :15.58   Mean   :14.585   Mean   : 7.823   Mean   :43.35  
 3rd Qu.:16.73   3rd Qu.:16.200   3rd Qu.: 8.740   3rd Qu.:45.87  
 Max.   :30.48

    arrests           countyid        density             popul        
 Min.   :   0.00   Min.   : 1001   Min.   :    0.08   Min.   :    141  
 1st Qu.:   0.00   1st Qu.:20105   1st Qu.:   17.95   1st Qu.:  13584  
 Median :   1.00   Median :36065   Median :   46.54   Median :  30514  
 Mean   :   6.07   Mean   :32922   Mean   :  261.87   Mean   :  96249  
 3rd Qu.:   3.00   3rd Qu.:48049   3rd Qu.:  117.55   3rd Qu.:  73037  
 Max.   :1105.00   Max.   :56045   Max.   :54058.77   Max.   :9127751  
    perc1019         perc2029        percblack          percmale    
 Min.   : 7.599   Min.   : 5.617   Min.   : 0.0000   Min.   :44.51  
 1st Qu.:13.965   1st Qu.:10.131   1st Qu.: 0.2503   1st Qu.:48.39  
 Median :15.031   Median :11.762   Median : 1.6710   Median :49.14  
 Mean   :15.125   Mean   :12.014   Mean   : 8.2211   Mean   :49.34  
 3rd Qu.:16.120   3rd Qu.:13.190   3rd Qu.: 9.1981   3rd Qu.:49.93  
 Max.   :30.485   Max.   :34.755   Max.   :86.2793   Max.   :61.64  
  rpcincmaint

Current number of rows (Counties) =  2197 


In [10]:
attach(cm)

The following objects are masked from countymurders:

    arrestrate, arrests, countyfips, countyid, density, execrate,
    execs, lpopul, murders, murdrate, perc1019, perc2029, percblack,
    percmale, popul, rpcincmaint, rpcpersinc, rpcunemins, statefips,
    year




In [19]:
c<-sum(murders==0)
cat("The number of counties that had zero murders in 1996 is ",c, "\n")
d<-sum(execs>0)
cat("The number of counties that had at least one execution is ", d, "\n")
cat("The largest number of executions is ", max(execs), "\n")

The number of counties that had zero murders in 1996 is  1051 
The number of counties that had at least one execution is  31 
The largest number of executions is  3 


   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
0.00000 0.00000 0.00000 0.01593 0.00000 3.00000 

In [22]:
m4=lm(murders~execs)
m4
summary(m4)
nrow(cm)


Call:
lm(formula = murders ~ execs)

Coefficients:
(Intercept)        execs  
      5.457       58.555  



Call:
lm(formula = murders ~ execs)

Residuals:
    Min      1Q  Median      3Q     Max 
-149.12   -5.46   -4.46   -2.46 1338.99 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)    
(Intercept)   5.4572     0.8348   6.537 7.79e-11 ***
execs        58.5555     5.8333  10.038  < 2e-16 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 38.89 on 2195 degrees of freedom
Multiple R-squared:  0.04389,	Adjusted R-squared:  0.04346 
F-statistic: 100.8 on 1 and 2195 DF,  p-value: < 2.2e-16


[1] 2197

#Linear regression model
##### **_$murders$_**= 5.457 + 58.555 **_$execs$_**

#####$n$ =2197
#####$R^{2}$ = 0.0439


The coefficient of $exec$ is statistically significant.

The estimated equation shows that if there is one more execution, it is associated with 58.555 more murders. Therefore it doesn't suggest a deterrent effect of capital punishmnet, but rather the opposite

The smallest number of murders that can be predicted by the equation is $5.457$ (the intercept) (but according to the dataset there are counties with zero murders)

The residual for a county with zero executions and zero murders is $-5.457$

A simple regression analysis is not well suited for determining  whether capital punishment has a deterrent effect on murders due to the following reasons.
1. The $R{^2}$ value is less than 5%. The executions do not really explain much of the variation in murders (Murders of a county can be a function of many other variables such as average income, unemployment, and many more).
2. Endogeneity issue : There can be an endogeneity issue. There might be causality in the other way (more murders cause executions) or there are other variables that are corelated with both these variables. 